app/template/page.client.tsx

```diff
@@ -0,0 +1,187 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import PageHeading from '../_components/utility/pageHeading';
+import getFromApiClientSide from './api';
+import layoutConfig from '../_config/layout';
+import GetColor from '../_hooks/colorSelector';
+import {
+  Alert,
+  AlertIcon,
+  Box,
+  Button,
+  Center,
+  Flex,
+  List,
+  ListItem,
+  Stack,
+  Text,
+  VStack,
+} from '@chakra-ui/react';
+
+export interface IServerDataSchema {
+  commodityName: string;
+  displayName: string;
+  type: string;
+  isRare: boolean;
+}
+
+interface Props {
+  serverData: IServerDataSchema[];
+}
+
+const PageClient: React.FC<Props> = ({ serverData }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+  const [clientResponse, setClientResponse] = useState<
+    IServerDataSchema[] | null
+  >([]);
+
+  const { handleSubmit } = useForm();
+
+  const onSubmit = async (): Promise<void> => {
+    setSubmitted(false);
+    setIsLoading(true);
+    setClientResponse([]);
+
+    const res = await getFromApiClientSide();
+
+    if (!res.ok) {
+      setClientResponse(null);
+      setSubmitted(true);
+      setIsLoading(false);
+      return;
+    }
+    const json = await res.json();
+    setClientResponse(json);
+    setSubmitted(true);
+    setIsLoading(false);
+  };
+
+  const displayResults = (clientData: IServerDataSchema[], label: string) => {
+    if (clientData.length === 0) {
+      return (
+        <Alert status="warning" borderRadius="md">
+          <AlertIcon />
+          No results found!
+        </Alert>
+      );
+    }
+    return (
+      <Stack
+        direction={{
+          base: 'column',
+          sm: 'column',
+          md: 'column',
+          lg: 'row',
+        }}
+        width="100%"
+        spacing={8}
+        fontSize="xl"
+      >
+        <Text>{label}</Text>
+        <List display="flex" gap={8}>
+          {clientData.map((item) => (
+            <ListItem key={item.commodityName}>{item.displayName}</ListItem>
+          ))}
+        </List>
+      </Stack>
+    );
+  };
+
+  return (
+    <Box flex="1" paddingX={2} width="100%">
```

Comment:
Can this be broken down any further? I noticed you did a PageHeader component which is awesome, really simplifies keeping layouts consistent, so could this potentially be broken down into a ContentArea component or something?

Reply:
It certainly could. So this is returning the results of the form. The first question is, are those results going to be displayed in a universal way across our pages? We may not be able to answer this now. In the commodities route, I kept the results in the route. Looking at how that data is displayed, can it be made generic enough to create a results component that can be reused?

I think this is better saved for another PR. Here, i tried to render the results in a simple manner just so devs can see that the fetch calls worked.